src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx

```diff
@@ -22,10 +61,170 @@ class Accretion extends Analyzer {
       Events.cast.by(SELECTED_PLAYER).spell(TALENTS.ERUPTION_TALENT),
       this.onCast,
     );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(this.currentUpheaval),
+      this.uphealvelCast,
+    );
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.SHIFTING_SANDS_BUFF),
+      this.onBuffApply,
+    );
+    this.addEventListener(Events.damage, this.onDamage);
+    this.addEventListener(Events.fightend, this.calcAccretionValue);
   }
 
   onCast() {
-    this.spellUsable.reduceCooldown(this.currentUpheaval, ACCRETION_CDR_MS);
+    this.effectiveUpheavelCDR += this.spellUsable.reduceCooldown(
+      this.currentUpheaval.id,
+      ACCRETION_CDR_MS,
+    );
+    this.totalEruptionCasts += 1;
+  }
+
+  onBuffApply() {
+    this.totalShiftingSandsApplications += 1;
+  }
+
+  uphealvelCast() {
+    const critChance = this.stats.currentCritPercentage;
+    const critMod = 1 + SANDS_OF_TIME_CRIT_MOD * critChance;
+    if (this.selectedCombatant.hasBuff(SPELLS.EBON_MIGHT_BUFF_PERSONAL.id)) {
+      this.ebonMightUpheavelExtention += EMPOWER_EXTENSION_MS * critMod;
+    }
+    this.totalUpheavelCasts += 1;
+  }
+
+  onDamage(event: DamageEvent) {
+    if (event.ability.guid === SPELLS.UPHEAVAL_DAM.id) {
+      this.totalUpheavelDamage += event.amount + (event.absorbed ?? 0);
+    }
+    if (event.ability.guid === TALENTS.EBON_MIGHT_TALENT.id) {
+      this.totalEbonMightDamage += event.amount + (event.absorbed ?? 0);
+    }
+    if (event.ability.guid === SPELLS.SHIFTING_SANDS_BUFF.id) {
+      this.totalShiftingSandsDamage += event.amount + (event.absorbed ?? 0);
+    }
+    this.totalDamageDone += event.amount + (event.absorbed ?? 0);
+  }
+
+  private calcAccretionValue() {
+    const EbonMightUptime = this.selectedCombatant.getBuffUptime(
+      SPELLS.EBON_MIGHT_BUFF_PERSONAL.id,
+    );
+    const additionalUpheavalCastsViaCdr = Math.floor(this.effectiveUpheavelCDR / (36 * 1000));
+
+    const avgUpheavalCastDamage = this.totalUpheavelDamage / this.totalUpheavelCasts;
+
+    this.accretionUpheavel = avgUpheavalCastDamage * additionalUpheavalCastsViaCdr;
+
+    this.upheavelPercentage = (this.accretionUpheavel / this.totalDamageDone) * 100;
+
+    const avgShiftingSandsDamage =
+      this.totalShiftingSandsDamage / this.totalShiftingSandsApplications;
+
+    this.accretionShiftingSands = avgShiftingSandsDamage * additionalUpheavalCastsViaCdr;
+
+    this.shiftingSandsPercentage = (this.accretionShiftingSands / this.totalDamageDone) * 100;
+
+    const cdrUpheavelExtention =
+      (this.ebonMightUpheavelExtention / this.totalUpheavelCasts) * additionalUpheavalCastsViaCdr;
+
+    this.accretionEbonMight = (this.totalEbonMightDamage / EbonMightUptime) * cdrUpheavelExtention;
+
+    this.ebonMightPercentage = (this.accretionEbonMight / this.totalDamageDone) * 100;
+
+    if (debug) {
```

Comment:
can / should this be removed?

Reply:
You thinking about the pencentages or the debugging ?

Reply:
debugging